smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,258 @@
+     * UserTuneElement userTuneElement = builder.build();
+     * <br>
+     * Values such as title, source, artist, length, source, track and uri can be set using their respective setters through {@link Builder}
+     * <br>
+     * Use {@link #buildEmptyUserTune()} only while disabling UserTune publish through {@link UserTuneManager#clearUserTune()}
+     */
+    public static final class Builder {
+        private String artist;
+        private Integer length;
+        private Integer rating;
+        private String source;
+        private String title;
+        private Integer track;
+        private URI uri;
+
+        private Builder() {
+        }
+
+        /**
+         * Artist is an optional element in UserTuneElement.
+         * @param artist
+         * @return Builder
+         */
+        public Builder setArtist(String artist) {
+            this.artist = artist;
+            return this;
+        }
+
+        /**
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            if (length > 0) {
+                this.length = length;
+                return this;
+            }
+            else {
+                throw new IllegalArgumentException("Length should be positive. In case of live streaming, length is expected to be null");
+            }
+        }
+
+        /**
+         * Rating is an optional element in UserTuneElement.
+         * The rating to be given is suppossed to be out of 10.
+         * <br>
+         * Method will throw an Exception if rating is less than 0 or greater than 10.
+         * @param rating
+         * @return Builder
+         */
+        public Builder setRating(int rating) {
+            if (rating <= 10 && rating >= 0) {
+                this.rating = rating;
+            } else {
+                throw new IllegalArgumentException("Rating tried to set was out of bounds");
+            }
+            return this;
+        }
+
+        /**
+         * Source is an optional element in UserTuneElement.
+         * @param source
+         * @return Builder
+         */
+        public Builder setSource(String source) {
+            this.source = source;
+            return this;
+        }
+
+        /**
+         * Title is a compulsory element in UserTuneElement.
+         * @param title
+         * @return Builder
+         */
+        public Builder setTitle(String title) {
+            this.title = title;
+            return this;
+        }
+
+        /**
+         * Track is an optional element in UserTuneElement.
+         * @param track
+         * @return Builder
+         */
+        public Builder setTrack(int track) {
+            if (track > 0) {
+                this.track = track;
+                return this;
+            }
+            else {
+                throw new IllegalArgumentException("Track number should be positive.");
```

Comment:
Same simplification to improve the readability of the method as above.

Reply:
I just consulted xep118, where track is specified as xs:string, so this needs to go away.